utils.py

```diff
@@ -0,0 +1,42 @@
+import numpy as np
+
+def get_index_dic(vocab, a):
```

Comment:
In general you want docstrings for utility functions. Otherwise, it's hard to know what they do. Sometimes code explains itself with variable names, but this is not the case with `a`.